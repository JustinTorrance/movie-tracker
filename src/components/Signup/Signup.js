import React, { Component } from 'react'
import { addUser } from '../../utils/apiCalls'
import { Redirect, Link } from 'react-router-dom'


export default class Signup extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      email: '',
      password: '',
      confirmPassword: '',
      duplicateUser: false,
      validUser: false,
      formComplete: false,
      passwordsMatch: true
    }
  }

  handleChange = async (e) => {
    const { name, value } = e.target
    await this.setState({
      [name]: value,
    })
    this.checkFormCompletion()
  }

  checkFormCompletion() {
    const { name, email, password, confirmPassword } = this.state
    this.setState({
      formComplete: (name !== '' && email !== '' && password !== '' && confirmPassword !== '')
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    const { name, email, password, confirmPassword } = this.state
    if (password !== confirmPassword) {
      this.setState({ duplicateUser: false, passwordsMatch: false })
      return
    }
    const data = await addUser({ name, email, password }) 
    if (data.error) {
      this.setState({ duplicateUser: true, passwordsMatch: true})
    } else {
      this.setState({ validUser: true})
    }
  }

  render() {
    const { name, email, password, confirmPassword, validUser, formComplete, duplicateUser, passwordsMatch } = this.state
    if (validUser) {
     return <Redirect to='/login' />
    }
    return(
      <div className="sign-up-container">
        <h1 className='form-title'>
          movie
          <img src='./images/movie_roll.svg' alt="movie roll" className='form-logo'></img>
          tracker
        </h1>
        <h2 className='login-title'>Sign Up</h2>
        <form className='login-form create-new-user-form' onSubmit={this.handleSubmit}>
          <input 
            onChange={this.handleChange}
            type='text'
            placeholder='name'
            name='name'
            value={name}
          />
          <input 
            onChange={this.handleChange}
            type='text'
            placeholder='email'
            name='email'
            value={email}
          />
          <input 
            onChange={this.handleChange}
            type='password'
            placeholder='password'
            name='password'
            value={password}
          />
          <input 
            onChange={this.handleChange}
            type='password'
            placeholder='confirm password'
            name='confirmPassword'
            value={confirmPassword}
          />
          <button disabled={!formComplete}>Submit</button>
          <Link className='form-link' to='/login' >Return to login</Link>
        </form>
        <h3 className={`duplicate-user-message ${duplicateUser && 'create-user-error'}`}>An account already exists with that email</h3>
        <h3 className={`passwords-message ${!passwordsMatch && 'create-user-error'}`}>Passwords must match</h3>
      </div>
    )
  }
}