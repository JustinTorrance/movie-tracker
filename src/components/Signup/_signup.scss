.create-new-user-form {
  button {
    &:disabled {
      opacity: 0.5;
    } 

    &:disabled:hover {
      cursor: default;
      transform: scale(1);
      background-color: rgb(42, 133, 168);
      color: white;
    }
  }
}

.duplicate-user-message,
.passwords-message {
  color: rgb(207, 43, 43);
  display: none;
  font-weight: 300;
  font-size: 1em;
  margin: 15px auto;
  text-align: center;
  width: 350px; 
}

.create-user-error {
  display: block;
}