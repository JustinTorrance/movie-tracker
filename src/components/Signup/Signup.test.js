import { shallow } from 'enzyme'
import Signup from './Signup'
import React from 'react'

describe('SignUp Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Signup />)
  })

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot if it renders the redirect', () => {
    wrapper.state().validUser = true
    expect(wrapper).toMatchSnapshot()
  })

  it('Should have a default state', () => {
    const expected = {
      name: '',
      email: '',
      password: '',
      confirmPassword: '',
      duplicateUser: false,
      validUser: false,
      formComplete: false,
      passwordsMatch: true
    }

    expect(wrapper.state()).toEqual(expected)
  })

  describe('handleChange', () => [

  ])

  describe('checkFormCompletion', () => {

  })

  describe('handleSubmit', () => {

  })
})