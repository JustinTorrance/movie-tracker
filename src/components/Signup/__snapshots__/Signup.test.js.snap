// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="form-title"
        >
          movie
          <img
            alt="movie roll"
            className="form-logo"
            src="./images/movie_roll.svg"
          />
          tracker
        </h1>,
        <h2
          className="login-title"
        >
          Sign Up
        </h2>,
        <form
          className="login-form create-new-user-form"
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
          />
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
          <input
            name="confirmPassword"
            onChange={[Function]}
            placeholder="confirm password"
            type="password"
            value=""
          />
          <button
            disabled={true}
          >
            Submit
          </button>
          <Link
            className="form-link"
            replace={false}
            to="/login"
          >
            Return to login
          </Link>
        </form>,
        <h3
          className="duplicate-user-message false"
        >
          An account already exists with that email
        </h3>,
        <h3
          className="passwords-message false"
        >
          Passwords must match
        </h3>,
      ],
      "className": "sign-up-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "movie",
            <img
              alt="movie roll"
              className="form-logo"
              src="./images/movie_roll.svg"
            />,
            "tracker",
          ],
          "className": "form-title",
        },
        "ref": null,
        "rendered": Array [
          "movie",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie roll",
              "className": "form-logo",
              "src": "./images/movie_roll.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "tracker",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "login-title",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />,
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />,
            <input
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirm password"
              type="password"
              value=""
            />,
            <button
              disabled={true}
            >
              Submit
            </button>,
            <Link
              className="form-link"
              replace={false}
              to="/login"
            >
              Return to login
            </Link>,
          ],
          "className": "login-form create-new-user-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "confirmPassword",
              "onChange": [Function],
              "placeholder": "confirm password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "disabled": true,
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Return to login",
              "className": "form-link",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Return to login",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "An account already exists with that email",
          "className": "duplicate-user-message false",
        },
        "ref": null,
        "rendered": "An account already exists with that email",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Passwords must match",
          "className": "passwords-message false",
        },
        "ref": null,
        "rendered": "Passwords must match",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="form-title"
          >
            movie
            <img
              alt="movie roll"
              className="form-logo"
              src="./images/movie_roll.svg"
            />
            tracker
          </h1>,
          <h2
            className="login-title"
          >
            Sign Up
          </h2>,
          <form
            className="login-form create-new-user-form"
            onSubmit={[Function]}
          >
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <input
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirm password"
              type="password"
              value=""
            />
            <button
              disabled={true}
            >
              Submit
            </button>
            <Link
              className="form-link"
              replace={false}
              to="/login"
            >
              Return to login
            </Link>
          </form>,
          <h3
            className="duplicate-user-message false"
          >
            An account already exists with that email
          </h3>,
          <h3
            className="passwords-message false"
          >
            Passwords must match
          </h3>,
        ],
        "className": "sign-up-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "movie",
              <img
                alt="movie roll"
                className="form-logo"
                src="./images/movie_roll.svg"
              />,
              "tracker",
            ],
            "className": "form-title",
          },
          "ref": null,
          "rendered": Array [
            "movie",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "movie roll",
                "className": "form-logo",
                "src": "./images/movie_roll.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "tracker",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "login-title",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                placeholder="name"
                type="text"
                value=""
              />,
              <input
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
              />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />,
              <input
                name="confirmPassword"
                onChange={[Function]}
                placeholder="confirm password"
                type="password"
                value=""
              />,
              <button
                disabled={true}
              >
                Submit
              </button>,
              <Link
                className="form-link"
                replace={false}
                to="/login"
              >
                Return to login
              </Link>,
            ],
            "className": "login-form create-new-user-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "confirmPassword",
                "onChange": [Function],
                "placeholder": "confirm password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "disabled": true,
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Return to login",
                "className": "form-link",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Return to login",
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "An account already exists with that email",
            "className": "duplicate-user-message false",
          },
          "ref": null,
          "rendered": "An account already exists with that email",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Passwords must match",
            "className": "passwords-message false",
          },
          "ref": null,
          "rendered": "Passwords must match",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`SignUp Component Should match the snapshot if it renders the redirect 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="form-title"
        >
          movie
          <img
            alt="movie roll"
            className="form-logo"
            src="./images/movie_roll.svg"
          />
          tracker
        </h1>,
        <h2
          className="login-title"
        >
          Sign Up
        </h2>,
        <form
          className="login-form create-new-user-form"
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
          />
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="text"
            value=""
          />
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
          <input
            name="confirmPassword"
            onChange={[Function]}
            placeholder="confirm password"
            type="password"
            value=""
          />
          <button
            disabled={true}
          >
            Submit
          </button>
          <Link
            className="form-link"
            replace={false}
            to="/login"
          >
            Return to login
          </Link>
        </form>,
        <h3
          className="duplicate-user-message false"
        >
          An account already exists with that email
        </h3>,
        <h3
          className="passwords-message false"
        >
          Passwords must match
        </h3>,
      ],
      "className": "sign-up-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "movie",
            <img
              alt="movie roll"
              className="form-logo"
              src="./images/movie_roll.svg"
            />,
            "tracker",
          ],
          "className": "form-title",
        },
        "ref": null,
        "rendered": Array [
          "movie",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie roll",
              "className": "form-logo",
              "src": "./images/movie_roll.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "tracker",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "login-title",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />,
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />,
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />,
            <input
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirm password"
              type="password"
              value=""
            />,
            <button
              disabled={true}
            >
              Submit
            </button>,
            <Link
              className="form-link"
              replace={false}
              to="/login"
            >
              Return to login
            </Link>,
          ],
          "className": "login-form create-new-user-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "confirmPassword",
              "onChange": [Function],
              "placeholder": "confirm password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "disabled": true,
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Return to login",
              "className": "form-link",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Return to login",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "An account already exists with that email",
          "className": "duplicate-user-message false",
        },
        "ref": null,
        "rendered": "An account already exists with that email",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Passwords must match",
          "className": "passwords-message false",
        },
        "ref": null,
        "rendered": "Passwords must match",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="form-title"
          >
            movie
            <img
              alt="movie roll"
              className="form-logo"
              src="./images/movie_roll.svg"
            />
            tracker
          </h1>,
          <h2
            className="login-title"
          >
            Sign Up
          </h2>,
          <form
            className="login-form create-new-user-form"
            onSubmit={[Function]}
          >
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="text"
              value=""
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <input
              name="confirmPassword"
              onChange={[Function]}
              placeholder="confirm password"
              type="password"
              value=""
            />
            <button
              disabled={true}
            >
              Submit
            </button>
            <Link
              className="form-link"
              replace={false}
              to="/login"
            >
              Return to login
            </Link>
          </form>,
          <h3
            className="duplicate-user-message false"
          >
            An account already exists with that email
          </h3>,
          <h3
            className="passwords-message false"
          >
            Passwords must match
          </h3>,
        ],
        "className": "sign-up-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "movie",
              <img
                alt="movie roll"
                className="form-logo"
                src="./images/movie_roll.svg"
              />,
              "tracker",
            ],
            "className": "form-title",
          },
          "ref": null,
          "rendered": Array [
            "movie",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "movie roll",
                "className": "form-logo",
                "src": "./images/movie_roll.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "tracker",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "login-title",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="name"
                onChange={[Function]}
                placeholder="name"
                type="text"
                value=""
              />,
              <input
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="text"
                value=""
              />,
              <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
              />,
              <input
                name="confirmPassword"
                onChange={[Function]}
                placeholder="confirm password"
                type="password"
                value=""
              />,
              <button
                disabled={true}
              >
                Submit
              </button>,
              <Link
                className="form-link"
                replace={false}
                to="/login"
              >
                Return to login
              </Link>,
            ],
            "className": "login-form create-new-user-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "confirmPassword",
                "onChange": [Function],
                "placeholder": "confirm password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "disabled": true,
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Return to login",
                "className": "form-link",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Return to login",
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "An account already exists with that email",
            "className": "duplicate-user-message false",
          },
          "ref": null,
          "rendered": "An account already exists with that email",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Passwords must match",
            "className": "passwords-message false",
          },
          "ref": null,
          "rendered": "Passwords must match",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
